source/core_mqtt_serializer.c

```diff
@@ -4093,35 +4223,44 @@ MQTTStatus_t MQTT_DeserializeAck( const MQTTPacketInfo_t * pIncomingPacket,
         LogError( ( "Remaining data of incoming packet is NULL." ) );
         status = MQTTBadParameter;
     }
+    /* Max packet size cannot be 0. */
+    else if( pConnectProperties->maxPacketSize == 0U )
+    {
+        LogError( ( "Max packet size cannot be 0." ) );
+        status = MQTTBadParameter;
+    }
+    else if( ( pIncomingPacket->remainingLength +
+               variableLengthEncodedSize( pIncomingPacket->remainingLength ) +
+               1U ) > pConnectProperties->maxPacketSize )
+    {
+        LogError( ( "Packet Size cannot be greater than max packet size." ) );
+        status = MQTTBadResponse;
+    }
     else
     {
         /* Make sure response packet is a valid ack. */
         switch( pIncomingPacket->type )
         {
-            case MQTT_PACKET_TYPE_CONNACK:
-                LogError( ( "Connack should be deserialized with MQTT_DeserializeConnAck" ) );
+            case MQTT_PACKET_TYPE_PUBACK:
+            case MQTT_PACKET_TYPE_PUBREC:
+            case MQTT_PACKET_TYPE_PUBREL:
+            case MQTT_PACKET_TYPE_PUBCOMP:
+                /* TODO: Add this section. */
```

Comment:
This should be fixed in a later PR, correct?

Reply:
Correct - when we add the mqtt publish APIs